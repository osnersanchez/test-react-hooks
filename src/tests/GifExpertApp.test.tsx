import React from 'react';
import '@testing-library/jest-dom';
import { shallow, ShallowWrapper } from 'enzyme';

import GitExpertApp from '../GifExpertApp';


describe('GitExpertApp test suite', () => {

  test('should render default template when default properties are set', () => {
    const wrapper: ShallowWrapper = shallow(<GitExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render default template when default properties are set', () => {
    const categories = ['One Push', 'Dragon Ball'];

    const wrapper: ShallowWrapper = shallow(<GitExpertApp defaultCategory={categories} />);
    expect(wrapper).toMatchSnapshot();
  });
});