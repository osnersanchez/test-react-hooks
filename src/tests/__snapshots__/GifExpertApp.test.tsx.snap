// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitExpertApp test suite should render default template when default properties are set 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
  </ol>
</Fragment>
`;

exports[`GitExpertApp test suite should render default template when default properties are set 2`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Push"
      key="One Push"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
